operator/pkg/helmreconciler/wait.go

```diff
@@ -279,18 +278,13 @@ func daemonsetsReady(daemonsets []*appsv1.DaemonSet) (bool, []string) {
 			notReady = append(notReady, "DaemonSet/"+ds.Namespace+"/"+ds.Name)
 		}
 		if ds.Spec.UpdateStrategy.Type == appsv1.RollingUpdateDaemonSetStrategyType {
-			maxUnavailable, err := intstr.GetValueFromIntOrPercent(
-				ds.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable,
-				int(ds.Status.DesiredNumberScheduled),
-				true)
-			if err != nil {
-				// set max unavailable to the number of desired replicas if the value is invalid
-				maxUnavailable = int(ds.Status.DesiredNumberScheduled)
-			}
-			expectedReady := int(ds.Status.DesiredNumberScheduled) - maxUnavailable
-			if !(int(ds.Status.NumberReady) >= expectedReady) {
+			if ds.Status.NumberReady != ds.Status.DesiredNumberScheduled {
 				scope.Infof("DaemonSet is not ready: %s/%s. %d out of %d expected pods are ready",
-					ds.Namespace, ds.Name, ds.Status.NumberReady, expectedReady)
+					ds.Namespace, ds.Name, ds.Status.NumberReady, ds.Status.UpdatedNumberScheduled)
+				notReady = append(notReady, "DaemonSet/"+ds.Namespace+"/"+ds.Name)
+			} else if ds.Status.DesiredNumberScheduled == 0 {
```

Comment:
Should this be `ds.Status.NumberReady`?

Reply:
Here is `else if`,  so it's equivalent to `ds.Status.NumberReady`.

I think DesiredNumberScheduled is better than NumberReady in explain the initialization status

Reply:
Can we reverse the condition here?

```
if desired <= 0 {
  log initializing.
} else if ds.Status.NumberReady != ds.Status.DesiredNumberScheduled {
  log some pod is not ready.
}
```

I saw cases where (desired = -1, ready = 0), so this logic make more sense to me.